### PR TITLE
Move asset list to home tab on small screens

### PR DESCRIPTION
**Pull request by Gudahtt**
Two tabs have been created on the home screen: 'Assets' and 'History'. This tabbed view is shown only on small screens (e.g. in the popup). The fullscreen view is unchanged.

The toggle-able left sidebar no longer exists, so some 'sidebar-left' specific code and styles have been removed. The button in the menu bar has been removed as well.

The 'History' title of the transaction history is now redundant when where are no pending transactions, so it as been conditionally hidden.

A passthrough for `data-testid` has been added to the Tab component for convenience in e2e tests.

### CODE REVIEW

**Comment by Gudahtt** [comment]
<details>
<summary>Screenshots:</summary>
Here is the new popup UI with tabs:

home-tabs

Here's what it looks like when scrolled down somewhat:

home-tabs-scrolled

Here's what the it looks like at its widest point (beyond which it goes to the fullscreen UI):

wide-home-tabs

</details>

**Comment by Gudahtt** [comment]
When you click on a token, it no longer brings you to the transaction history for that token like it used to. I opened #8266 in preparation for updating the Home component to automatically switch to the `History` tab when a token was selected, but after using it locally I'm not sure I like the idea 🤔. Either way the interaction is _not great_, but the automatic tab switching does feel a little more weird.

File: ui/app/components/app/asset-list/asset-list.component.js [hunk 1]
```diff
@@ -104,7 +89,6 @@ export default class AssetList extends Component {
                 name: 'Clicked Token',
               },
             })
-            selectedTokenAddress !== tokenAddress && sidebarOpen && hideSidebar()
           }}
         />
         {this.renderAddToken()}
```
Comment (danjm) [inline ui/app/components/app/asset-list/asset-list.component.js:94]: okay, these file changes just remove the triggering of sidebar hiding from the asset list component. Looks good, makes sense, and explains the below change to the container

File: ui/app/components/app/menu-bar/menu-bar.component.js [hunk 2]
```diff
@@ -10,39 +10,14 @@ export default class MenuBar extends PureComponent {
...
-                  name: 'Opened Hamburger',
-                },
-              })
-              sidebarOpen ? hideSidebar() : showSidebar()
-            }}
-          />
-        </Tooltip>
         <SelectedAccount />
 
         <Tooltip
```
Comment (danjm) [inline ui/app/components/app/menu-bar/menu-bar.component.js:23]: removes the button for opening the sidebar in popup view. looks good, makes sense

File: ui/app/components/app/sidebars/sidebar.component.js [hunk 3]
```diff
@@ -35,8 +33,6 @@ export default class Sidebar extends Component {
     const { type, sidebarProps = {} } = this.props
     const { transaction = {} } = sidebarProps
     switch (type) {
-      case WALLET_VIEW_SIDEBAR:
-        return <WalletView responsiveDisplayClassname="sidebar-right" />
       case 'customize-gas':
```
Comment (danjm) [inline ui/app/components/app/sidebars/sidebar.component.js:36]: no need for the sidebar component to know about the wallet view if we are no longer showing it in sidebar. Looks good, makes sense

File: ui/app/components/app/transaction-list/transaction-list.component.js [hunk 4]
```diff
@@ -100,9 +100,13 @@ export default class TransactionList extends PureComponent {
           )
         }
         <div className="transaction-list__completed-transactions">
-          <div className="transaction-list__header">
-            { t('history') }
-          </div>
+          {
+            pendingLength > 0 && (
```
Comment (danjm) [inline ui/app/components/app/transaction-list/transaction-list.component.js:104]: Okay, so the history title is only needed when the "History" tab needs to be divided into pending and history. Makes sense

File: ui/app/components/app/wallet-view/wallet-view.component.js [hunk 5]
```diff
@@ -18,25 +18,16 @@ export default class WalletView extends Component {
...
-      sidebarOpen,
-      hideSidebar,
-      history,
-    } = this.props
+    const { history } = this.props
     history.push(CONNECTED_ROUTE)
-    if (sidebarOpen) {
-      hideSidebar()
-    }
   }
```
Comment (danjm) [inline ui/app/components/app/wallet-view/wallet-view.component.js:32]: wallet view no longer interacts with a sidebar, doesn't need to know about it :ok_hand:, this implies below container changes

File: ui/app/pages/routes/routes.component.js [hunk 6]
```diff
@@ -250,7 +237,6 @@ export default class Routes extends Component {
           transitionName={sidebarTransitionName}
           type={sidebarType}
           sidebarProps={sidebar.props}
-          onOverlayClose={sidebarOnOverlayClose}
         />
         <NetworkDropdown
           provider={provider}
```
Comment (danjm) [inline ui/app/pages/routes/routes.component.js:242]: :ok_hand:

File: ui/app/components/ui/tabs/tab/tab.component.js [hunk 7]
```diff
@@ -24,6 +32,7 @@ const Tab = (props) => {
 
 Tab.propTypes = {
   className: PropTypes.string,
+  'data-testid': PropTypes.string,
   isActive: PropTypes.bool, // required, but added using React.cloneElement
   name: PropTypes.string.isRequired,
   onClick: PropTypes.func,
```
Comment (danjm) [inline ui/app/components/ui/tabs/tab/tab.component.js:38]: :ok_hand:

File: ui/app/pages/home/home.component.js [hunk 8]
```diff
@@ -179,18 +181,34 @@ export default class Home extends PureComponent {
...
+                        <Tab className="home__tab" data-testid="home__history-tab" name="History">
+                          <TransactionList />
+                        </Tab>
+                      </Tabs>
                     </div>
-                    <TransactionList />
-                  </div>
-                </>
+                  )
               )
```
Comment (danjm) [inline ui/app/pages/home/home.component.js:212]: refactor `isWideViewport` logic and add new tabs to home screen, :ok_hand:

**Review by danjm — APPROVED** [review]
Reviewed every file, everything looks good. My comments are just what I wrote on the important files after reading them and understanding them. No action is needed.

**Comment by Gudahtt** [comment]
I'm not happy with the UX of this new home page - it's ambiguous whether this is a step forward or not. But it's a step toward designs that _are_ better at least, so I'll merge this for now and we can improve it later.

**Pull request merged**